CNN.diff

```diff
@@ -0,0 +1,317 @@
+                 probs,  predicted = torch.max(outputs, 1)
+ 
+@@ -174,7 +186,7 @@ class IID_model():
+     def calculate_probs(self, data=None):
+         
+         n_features = self.n_features
+-        test_dataset = SequenceDataset(self.sequence_file, k=self.k, transform=None, GT_file=self.GT_file, reduce=self.reduce)
++        test_dataset = SequenceDataset(self.sequence_file, k=self.k, transform=None, GT_file=self.GT_file, reduce=self.reduce, representation=self.model_type)
+         test_dataloader = DataLoader(test_dataset, 
+                              batch_size=self.batch_sz,
+                              shuffle=False,
+@@ -185,7 +197,10 @@ class IID_model():
+         with torch.no_grad():
+             self.net.eval()
+             for test in test_dataloader:
+-                kmers = test['kmer'].view(-1, 1, n_features).type(dtype)
++                if self.model_type not in ('conv'):
++                    kmers = test['kmer'].view(-1, 1, n_features).type(dtype)
++                else:
++                    kmers = test['kmer'][:, np.newaxis, ...].type(dtype)
+ 
+                 #calculate the prediction by running through the network
+                 outputs, logits = self.net(kmers)
+diff --git a/src/utils.py b/src/utils.py
+index 0ef4a52..ec23302 100644
+--- a/src/utils.py
++++ b/src/utils.py
+@@ -3,7 +3,7 @@ sys.path.append('src/')
+ import pyximport 
+ pyximport.install()
+ 
+-from kmers import kmer_counts
++from kmers import kmer_counts, cgr
+ 
+ import random, itertools
+ import numpy as np
+@@ -185,7 +185,7 @@ def SummaryFasta(fname, GT_file=None):
+     return names, lengths, ground_truth, cluster_dis
+ 
+ 
+-def kmersFasta(fname, k=6, transform=None, reduce=False):
++def kmersFasta(fname, k=6, transform=None, reduce=False, representation='vector'):
+     lines = list()
+     seq_id = ""
+     names, kmers = [], []
+@@ -203,8 +203,12 @@ def kmersFasta(fname, k=6, transform=None, reduce=False):
+                     transform(seq)
+                     
+                 counts = np.ones(4**k, dtype=np.int32)
+-                kmer_counts(seq, k, counts)
+-                #cgr(seq, k, counts)
++                if representation == 'vector':
++                    kmer_counts(seq, k, counts)
++                else:
++                    cgr(seq, k, counts)
++                    counts = counts.reshape((2**k, 2**k))
++
+                 kmers.append(counts / np.sum(counts))
+                 
+                 lines = []
+@@ -220,19 +224,24 @@ def kmersFasta(fname, k=6, transform=None, reduce=False):
+         transform(seq)
+         
+     counts = np.ones(4**k, dtype=np.int32)
+-    kmer_counts(seq, k, counts)
+-    #cgr(seq, k, counts)
++    if representation == 'vector':
++        kmer_counts(seq, k, counts)
++    else:
++        cgr(seq, k, counts)
++        counts = counts.reshape((2**k, 2**k))
++
+     kmers.append(counts / np.sum(counts))
+-    
++
++    result_array = np.array(kmers)
+     if reduce:
```

Comment:
Note that 'reduce' cannot be used with 'fgr'. We need to make sure that this does not happen